lib/src/arithmetic/ripple_carry_adder.dart

```diff
@@ -19,10 +19,28 @@ import 'package:rohd_hcl/rohd_hcl.dart';
 class RippleCarryAdder extends Adder {
   /// Constructs an n-bit adder based on inputs List of inputs.
   RippleCarryAdder(super.a, super.b, {super.name = 'ripple_carry_adder'}) {
+    final adder = RippleCarryAdderC(a, b, Const(0));
+    sum <= adder.sum;
+  }
+  
+}
+/// An [RippleCarryAdderC] is a digital circuit used for binary addition with 
+/// exposed carry signals.
+/// It consists of a series of full adders connected in a chain, with the carry
+/// output of each adder linked to the carry input of the next one. Starting
+/// from the least significant bit (LSB) to most significant bit (MSB), the
+/// adder sequentially adds corresponding bits of two binary numbers.
+class RippleCarryAdderC extends Adder {
```

Comment:
Why don't we just make `carryIn` an optional input on the same `RipplerCarryAdder`, then if it's `null` we just put `Const(0)` in its place?

Might also be good to update the configurator for this adder appropriately

Reply:
Would be nice to get an example of optional Logic interface signal.

Reply:
in the shift register, enable and reset are optional ports:
https://github.com/intel/rohd-hcl/blob/main/lib/src/shift_register.dart

Reply:
Done. Removed RippleCarryAdderC